src/main/java/org/folio/innreach/util/KafkaUtil.java

```diff
@@ -0,0 +1,114 @@
+package org.folio.innreach.util;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.folio.innreach.batch.contribution.listener.ContributionExceptionListener;
+import org.folio.innreach.domain.dto.folio.inventorystorage.InstanceIterationEvent;
+import org.folio.innreach.external.exception.ServiceSuspendedException;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.util.backoff.BackOff;
+import org.springframework.util.backoff.FixedBackOff;
+
+import java.net.SocketTimeoutException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG;
+
+@Log4j2
+@Setter
+@RequiredArgsConstructor
+public class KafkaUtil {
+
+  private static Map<String, ConcurrentMessageListenerContainer<String, InstanceIterationEvent>> consumersMap =
+    new HashMap<>();
+
+
+  private final Map<String,Object> consumerProperties;
+  private final String topic;
+  private final Deserializer<String> keyDeserializer;
+  private final Deserializer<InstanceIterationEvent> valueDeserializer;
+
+  private final Long interval;
+
+  private final Long maxAttempts;
+
+  private final int THREAD_SIZE = 2;
+
+  private final ContributionExceptionListener contributionExceptionListener;
+
+  public DefaultErrorHandler errorHandler() {
+
+    BackOff fixedBackOff = new FixedBackOff(interval, maxAttempts);
+    DefaultErrorHandler errorHandler = new DefaultErrorHandler((consumerRecord, exception) -> {
+      // logic to execute when all the retry attempts are exhausted
+      ConsumerRecord<String, InstanceIterationEvent> record = (ConsumerRecord<String, InstanceIterationEvent>) consumerRecord;
+      contributionExceptionListener.logWriteError(exception, record.value().getInstanceId());
+      log.info("Stopping consumer topic: {} after retry exhaustion", consumerRecord.topic());
+      stopConsumer(consumerRecord.topic());
+    }, fixedBackOff);
+    errorHandler.addRetryableExceptions(ServiceSuspendedException.class);
+    errorHandler.addRetryableExceptions(SocketTimeoutException.class);
+    return errorHandler;
+  }
+
+  public void startOrCreateConsumer(Object messageListner) {
```

Comment:
For methods that attempt to do something (here start if it isn't already started) I like to prepend a `try` on the method name. So here I think this method should simply be `tryStart`. I think the word `consumer` is confusing/misleading here. We're basically starting the container which starts one or more consumers. If we rename the class as I recommend above, I think `tryStart` is clear.